src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -110,7 +110,7 @@ await python37GetVersionTask
             if (python36_37worker != null)
             {
                 SetWorkerPathIfNeeded(setWorkerExecutable, python36_37worker.ExecutablePath);
-                ColoredConsole.WriteLine(AdditionalInfoColor(string.Format(infoVersionSelectMessage, python36_37worker.ExecutablePath, python36_37worker.Version)));
+                ColoredConsole.WriteLine(AdditionalInfoColor(string.Format(infoVersionSelectMessage, python36_37worker.Version, python36_37worker.ExecutablePath)));
```

Comment:
Sorry, I should've test the docker init scenario in the last version.

This line is printing out twice when doing docker init or build-native-deps.
I think it we should remove this line and would be sufficient to only print the python version on **func host start** [PythonHelpers.cs:177](https://github.com/Azure/azure-functions-core-tools/pull/1614/files#diff-88f494e2db490871070e912928cacfbcR177)

```
Found Python version 3.6.8 (python).
Writing .gitignore
Writing host.json
Writing local.settings.json
Writing C:\Users\hazeng\Projects\Functions\Python36Docker\.vscode\extensions.json
Found Python version 3.6.8 (python).
Writing Dockerfile
Writing .dockerignore
```

Reply:
This function also has other print statements that may be printed twice during above.

I think we should avoid calling this function twice. Will take a look

Reply:
It seems to me that this method is taking on too many roles. 
It's resolving the correct `WorkerLanguageVersionInfo` for the current environment, setting an environment variable that's needed for the worker, communicate the selection to the user, and throwing if the state is invalid based on few different options.

That could be broken down into multiple different methods that each take one role, and we can leave the orchasteration of these methods to the upstream caller. 

Something like `ResolveWorkerLanguageVersionInfo()` that just returns the right `WorkerLanguageVersionInfo`. An `AssertPythonVersion()` that will throw if the state is invalid. Maybe a `SetWorkerEnvironmentVariable` or `EnsureWorkerEnvironment()`, that initialized the environment, and the confirmation of the selection could be explicit by code paths that need it.

Reply:
Thanks @ahmelsayed, that does sounds like a cleaner approach. I will update the PR.